src/explore-education-statistics-common/src/components/form/__tests__/Form.test.tsx

```diff
@@ -141,6 +142,38 @@ describe('Form', () => {
     });
   });
 
+  test('calls `onChange` handler as form values are updated', async () => {
+    const handleChange = jest.fn();
+
+    const { user } = render(
+      <FormProvider
+        initialValues={{
+          firstName: 'Firstname',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().defined(),
+        })}
+      >
+        <Form id="test-form" onChange={handleChange} onSubmit={jest.fn()}>
+          The form
+          <FormFieldTextInput label="first name" name="firstName" />
+          <button type="submit">Submit</button>
+        </Form>
+      </FormProvider>,
+    );
+
+    const firstName = 'first name';
+
+    await user.type(
+      screen.getByRole('textbox', { name: 'first name' }),
+      firstName,
+    );
+
+    await waitFor(() => {
+      expect(handleChange).toHaveBeenCalledTimes(firstName.length);
+    });
```

Comment:
Can we do a final check that the `handleChange` has been called with the expected value? Probably makes sense to do this after the wait.